### PR TITLE
KPIV-17: add join-options to conference command and add wait-mod

### PR DESCRIPTION
**Pull request by icehess**
This updates local join conference command to set member join options. For remote conference, current discovery request will be pass along in SIP Headers (if any) to let remote conference joins the member the original discovery request parameters.

Also updates build profile config to ensure some conference flags or profile parameters base on conference configuration (or discovery req while creating the conference session).

### CODE REVIEW

File: src/conf_participant.erl [hunk 1]
```diff
@@ -590,15 +596,23 @@ name_pronounced_headers({_, AccountId, MediaId}) ->
...
+              ,{<<"Profile">>, get_profile_name(Conference)}
+              ,{<<"Reinvite">>, 'false'}
+              ,{<<"Account-ID">>, kapps_call:account_id(Call)}
+              ,{<<"Join-Options">>
+               ,kz_json:from_list(
+                  [{<<"Mute">>, is_muted(Conference)}
+                  ,{<<"Deaf">>, is_deaf(Conference)}
+                  ,{<<"Moderator">>, kapps_conference:moderator(Conference)}
+                  ,{<<"Video-Mute">>, is_video_muted(Conference)}
+                  ,{<<"Endconf">>, kapps_conference:endconf(Conference)}
```
Comment (lazedo) [inline src/conf_participant.erl]:
```suggestion
                  ,{<<"End-Conference">>, kapps_conference:endconf(Conference)}
```

File: src/conf_participant.erl [hunk 2]
```diff
@@ -590,34 +611,51 @@ name_pronounced_headers({_, AccountId, MediaId}) ->
...
 is_video_muted(Conference) ->
     kapps_conference:participant_join_video_muted(Conference).
 
+-spec get_last_member_endconf(kapps_conference:conference()) -> kz_term:api_boolean().
+get_last_member_endconf(Conference) ->
+    case kapps_conference:discovery_request(Conference) of
+        'undefined' ->
+            'undefined';
+        DiscoveryJObj ->
+            kz_json:is_true(<<"Last-Member-Endconf">>, DiscoveryJObj, 'undefined')
```
Comment (lazedo) [inline src/conf_participant.erl]: `Last-Member-Endconf` does not express what the action is for `endconf`. a more appropriate is `End-Conference-On-Leave`, any member with this `flag` that leaves the conference will end the conference..
there is another flag `mandatory_member_endconf` that behaves like `Last-Member-Endconf` (I would change it to `End-Conference-On-Last-Member-Leave`) 
```
		/* End conference when any member with "endconf" flag disconnects */
		if (conference_utils_member_test_flag(member, MFLAG_ENDCONF)) {
			conference_utils_set_flag_locked(conference, CFLAG_DESTRUCT);
		}

		/* End conference only if all mandatory members have disconnected */
		if (conference_utils_member_test_flag(member, MFLAG_MANDATORY_MEMBER_ENDCONF)) {
			if (!--conference->end_count) {
				conference->endconference_time = switch_epoch_time_now(NULL);
			}
		}
```

Comment (icehess) [reply src/conf_participant.erl]: Ooh, spicy! I'll add both in that case. I didn't dive into fs code, just read the doc and i think it said `endconf` behave like mandatory members. guess their doc is out of date.

Comment (icehess) [reply src/conf_participant.erl]: Any objection to drop Confernce from the name? `End-On-Last-Member-Leave` and `End-On-Leave`.

**Pull request merged**